### PR TITLE
LPS-194385 LPS-201251 Migrate CustomViewsControls.js to Clay Picker

### PR DESCRIPTION
**Pull request by thektan**
**Story:** https://liferay.atlassian.net/browse/LPS-194385
**Subtask:** https://liferay.atlassian.net/browse/LPS-201251

This updates the Custom Views selector to use Clay Picker.

**Dev Note:**
- I introduced a new constant `DEFAULT_VIEW_ID` in order to be used for the Clay Picker `<Option>` `key` prop and the `selectedKey` to show which option was active.

## Demo

https://github.com/liferay-frontend/liferay-portal/assets/4496722/b5a9082b-7944-4976-81c3-1bc85775c0a5

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/CustomViewsControls.js [hunk 1]
```diff
@@ -15,6 +16,30 @@ import FrontendDataSetContext from '../../FrontendDataSetContext';
 import ViewsContext from '../../views/ViewsContext';
 import {VIEWS_ACTION_TYPES} from '../../views/viewsReducer';
 
+const DEFAULT_VIEW_ID = 'DEFAULT_VIEW';
+
+const CustomViewsControlsTrigger = React.forwardRef(
+	({triggerLabel, viewUpdated, ...otherProps}, ref) => (
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/CustomViewsControls.js:22]: [Interesting API](https://github.com/liferay/clay/blob/master/packages/clay-core/src/picker/Picker.tsx#L57-L63) . @matuzalemsteles is there a reason to require a `ref`? For seemingly the same purpose, a component prop, we used [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSetContext.tsx#L71) `React.ComponentType<any>`.

**Review by markocikos — APPROVED** [review]
@thektan LGTM! :sparkles:

**Comment by thektan** [comment]
ci:test:relevant

**Comment by JerryAdrianoo** [comment]
Hey @thektan, as you said, these changes require test fixes. So I'll start this task and send the changes.

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/CustomViewsControls.js (hunk 1 above)
Comment (matuzalemsteles) [reply modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/CustomViewsControls.js:22]: Hi @markocikos, In fact, this type is wrong, we just have to `React.ForwardRefExoticComponent`, we need the `ref` to calculate the menu positioning.

**Comment by JerryAdrianoo** [comment]
Hey, @thektan can you invite me as collaborator of your repository? I can't send the changes 😢

**Comment by JerryAdrianoo** [comment]
And also I found this bug when I was testing.
Can you take a look, please?

https://github.com/liferay-frontend/liferay-portal/assets/99041514/c30e8a5a-4d7b-4690-b7cf-607088a32bb7

**Comment by thektan** [comment]
@JerryAdrianoo Sent an invite!

For the bug, it's a bit hard for me to tell what the issue is. Is the issue the cell borders appearing while hovering over a row after you refresh the page?

**Comment by JerryAdrianoo** [comment]
@thektan In this video if you see, when the mouse over a row, the eye icon appears above the column line.
Well, for me, looks like a UX inconsistency.

**Comment by thektan** [comment]
@JerryAdrianoo Ah I see. This can be fixed in a separate ticket, since it doesn't seem related to the changes in this PR.

**Comment by markocikos** [comment]
@JerryAdrianoo @thektan Yeah, that behavior is unrelated to this PR. It's not a bug, it's expected behavior, but it's admittedly not a great UX. It's a result of the fact that quick menu had fixed width, while columns are variable. If there are a lot of columns, the quick menu clips over the column, and it's looks pretty buggy. @ugeortiz do you think we can improve this?

**Comment by JerryAdrianoo** [comment]
ci:test:relevant

**Comment by JerryAdrianoo** [comment]
ci:test:relevant

**Comment by JerryAdrianoo** [comment]
I'll run ci:relevant again because doesn't return any run.
image

**Comment by JerryAdrianoo** [comment]
ci:test:relevant

**Comment by JerryAdrianoo** [comment]
@thektan Failures are unrelated :)
[Test Comparison](https://testray.liferay.com/home/-/testray/runs/compare?formDate=1700704129957&testrayRunIdA=2463383875&testrayRunIdB=2463872428)

**Comment by thektan** [comment]
@JerryAdrianoo Thanks!

**Comment by thektan** [comment]
ci:forward:force

**Comment by markocikos** [comment]
ci:forward:force

**Pull request closed without merging**